platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -28,6 +30,35 @@
 
 GET_HWSKU_CMD = "sonic-cfggen -d -v DEVICE_METADATA.localhost.hwsku"
 
+PLATFORM_ROOT = '/usr/share/sonic/device'
+EEPROM_CACHE_ROOT = '/var/cache/sonic/decode-syseeprom'
+EEPROM_CACHE_FILE = 'syseeprom_cache'
+
+HWMGMT_SYSTEM_ROOT = '/var/run/hw-management/system/'
+
+REBOOT_CAUSE_ROOT = HWMGMT_SYSTEM_ROOT
+
+REBOOT_CAUSE_POWER_LOSS_FILE = 'reset_main_pwr_fail'
+REBOOT_CAUSE_THERMAL_OVERLOAD_ASIC_FILE = 'reset_asic_thermal'
+REBOOT_CAUSE_WATCHDOG_FILE = 'reset_hotswap_or_wd'
+REBOOT_CAUSE_MLNX_FIRMWARE_RESET = 'reset_fw_reset'
+
+REBOOT_CAUSE_FILE_LENGTH = 1
+
+REBOOT_DUE_TO_THIS_FILE = '1'
+
+CPLD_VERSION_ROOT = HWMGMT_SYSTEM_ROOT
+
+CPLD1_VERSION_FILE = 'cpld1_version'
+CPLD2_VERSION_FILE = 'cpld2_version'
+CPLD_VERSION_MAX_LENGTH = 4
+
+FIRMWARE_VERSION_COMMAND = 'mlxfwmanager --query'
+BIOS_VERSION_COMMAND = 'dmidecode -t 11'
+
+PREVIOUS_REBOOT_CAUSE_FILE = "/host/reboot-cause/previous-reboot-cause.txt"
+PREVIOUS_REBOOT_CAUSE_MAX_LENGTH = 100
+
 # magic code defnition for port number, qsfp port position of each hwsku
 # port_position_tuple = (PORT_START, QSFP_PORT_START, PORT_END, PORT_IN_BLOCK, EEPROM_OFFSET)
 hwsku_dict = {'ACS-MSN2700': 0, "LS-SN2700":0, 'ACS-MSN2740': 0, 'ACS-MSN2100': 1, 'ACS-MSN2410': 2, 'ACS-MSN2010': 3, 'ACS-MSN3700': 0, 'ACS-MSN3700C': 0, 'Mellanox-SN2700': 0, 'Mellanox-SN2700-D48C8': 0}
```

Comment:
why is there a difference? where does it come from? could it be changed in the future and can be ready somehow?

Reply:
This file contains the reboot reason when the switch is rebooted by a user. It is provided by SONiC and different from those files provided by hw-management.
The logic can be found in DUT's /etc/rc.local.
I've changed the name convention in order to be aligned with other REBOOT_CAUSE_XXX variables.